### PR TITLE
feat(clients): create BaseClient

### PR DESCRIPTION
**Pull request by Mesteery**

### CODE REVIEW

File: src/errors/messages.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
+const _messages = {
```
Comment (B-Derouet) [inline src/errors/messages.ts:1]: Why not use an Enum here?

File: src/clients/BaseClient.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+      callback(...args);
+    });
+
+    this.immediates.add(immediate);
+
+    return immediate;
+  }
+
+  public setTimeout(callback: (...args: unknown[]) => void, ms: number, ...args: unknown[]): NodeJS.Timeout {
```
[the remark this answers is not in the document]
Comment (Mesteery) [reply src/clients/BaseClient.ts:34]: Okay, can you make a suggestion ?

File: src/errors/messages.ts (hunk 1 above)
Comment (Mesteery) [reply src/errors/messages.ts:1]: Because an `enum` isn't appriopriate here.

File: src/errors/index.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  args.unshift(message);
+
+  return args.join(' ');
+}
+
+const _Error = makeError(Error);
+const _TypeError = makeError(TypeError);
+const _RangeError = makeError(RangeError);
+
+export { _Error as Error, _TypeError as TypeError, _RangeError as RangeError };
```
Comment (Mesteery) [inline src/errors/index.ts:40]: Someone have better solution ?

**Comment by Mesteery** [comment]
(don't delete the branch)

**Pull request merged**